Heisen test failure

As seen when running sklearn tests with joblib 0.8.1:

```
======================================================================
FAIL: Input is mutable when using the threading backend
----------------------------------------------------------------------
Traceback (most recent call last):
File "/home/travis/anaconda/envs/testenv/lib/python2.6/site-packages/nose/case.py", line 197, in runTest
self.test(*self.arg)
File "/home/travis/build/scikit-learn/scikit-learn/sklearn/externals/joblib/test/test_parallel.py", line 148, in test_increment_input_with_threads
nose.tools.assert_equal(a, [5])
AssertionError: [4] != [5]
>>  raise self.failureException, \
(None or '%r != %r' % ([4], [5]))
```

--- a/joblib/parallel.py
+++ b/joblib/parallel.py
@@ -103,10 +103,10 @@ class SafeFunction(object):
 
 
 ###############################################################################
-def delayed(function, check_picklability=True):
+def delayed(function, check_pickle=True):
     """Decorator used to capture the arguments of a function.
 
-    Pass `check_picklability=False` when:
+    Pass `check_pickle=False` when:
 
     - performing a possibly repeated check is too costly and has been done
       already once outside of the call to delayed.
@@ -116,7 +116,7 @@ def delayed(function, check_picklability=True):
     """
     # Try to pickle the input function, to catch the problems early when
     # using with multiprocessing:
-    if check_picklability:
+    if check_pickle:
         pickle.dumps(function)
 
     def delayed_function(*args, **kwargs):

--- a/joblib/test/test_parallel.py
+++ b/joblib/test/test_parallel.py
@@ -146,7 +146,7 @@ def test_mutate_input_with_threads():
     """Input is mutable when using the threading backend"""
     q = Queue(maxsize=5)
     Parallel(n_jobs=2, backend="threading")(
-        delayed(q.put, check_picklability=False)(1) for _ in range(5))
+        delayed(q.put, check_pickle=False)(1) for _ in range(5))
     nose.tools.assert_true(q.full())
 
 

